src/app/recipe-list/recipe-list.component.ts

```diff
@@ -15,24 +14,25 @@ export class RecipeListComponent implements OnInit {
   recipes$: Observable<Recipe[]> = this._recipeFacade.recipes$;
   loading$: Observable<boolean> = this._recipeFacade.loading$;
 
-  constructor(
-    private _recipeService: RecipeService,
-    private _router: Router,
-    private _recipeFacade: RecipeFacade
-  ) {}
+  constructor(private _router: Router, private _recipeFacade: RecipeFacade) {}
 
   ngOnInit(): void {
     this._recipeFacade.getAllRecipes();
   }
 
   filterRecipes(value: string): void {
     if (!value) {
-      this.recipes$ = this._recipeService.recipes$;
+      this.recipes$ = this._recipeFacade.recipes$;
     } else {
       value = value.toLowerCase();
-      this.recipes$ = this._recipeService.recipes$!.pipe(
-        map((recipes) =>
-          recipes.filter((recipe) => recipe.name?.includes(value))
+      this.recipes$ = this._recipeFacade.recipes$!.pipe(
+        map((recipes: Recipe[]): Recipe[] =>
+          recipes.filter(
+            (recipe: Recipe) =>
+              recipe.ingredients.filter((ingredient: Ingredient) =>
+                ingredient.name.includes(value)
+              ).length || recipe.name.includes(value)
+          )
```

Comment:
mógłbyś tę logikę przenieść do store'a, stworzyć klucz "filteredRecipes", akcję "filterRecipes", której reducer by wykonywał to całe filtrowanie i przypisywał do filteredRecipes